db/migrate/20150914100056_create_default_project_roles.rb

```diff
@@ -0,0 +1,19 @@
+class CreateDefaultProjectRoles < ActiveRecord::Migration
+  def change
+
+    #Creating default project role relations for non Viewers
+    User.where.not(role_id: Role::VIEWER.id).each { |user|
+      Project.find_each { |project|
+
+        if !UserProjectRole.exists?(project: project, user: user)
+          if user.role_id == Role::ADMIN.id
+            UserProjectRole.create!(project: project, user: user, role_id: ProjectRole::ADMIN.id)
+          elsif user.role_id == Role::DEPLOYER.id
+            UserProjectRole.create!(project: project, user: user, role_id: ProjectRole::DEPLOYER.id)
+          end
+        end
+      }
+    }
```

Comment:
You're absolutely right!

This script was the first thing I wrote, following the database model changes. Initially, I was thinking that system-level admins wouldn't have admin rights over the projects, that only super-admins and project-level admins would have it. So, my intention was to give admin rights to all projects to all current system-level admins, to avoid stripping them out of admin rights at the project level. After understanding better the requirements and how Samson works, I went back on this assumption and kept the admin rights for system-level admins at the project level. I should have come back and review this script. My bad.

Regarding the Deployer role, you're right, it makes no sense. Again, initially I was assuming that we also wanted to restrict system level deployers from being able to deploy any project, and start managing this permissions on a more granular basis. But after discussing this across the team here, I realised that a system-level Deployer should be able to deploy any project, which makes this part of the script unnecessary.

This being said, I'll remove the script. Thanks for the heads up.